From db568345911e635b7e8beeaa078ad242050386ee Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo.salveti@canonical.com>
Date: Wed, 16 Sep 2015 16:33:03 +0200
Subject: [PATCH] Create disk/by-partlabel links for mmcblk partitions

Android based kernel exports the uevent property PARTNAME, which can be
used to find out at run time the named partitions (e.g. boot) for the
device.

This is specially useful for the Touch based images and flash-kernel,
to automatically update the kernel by writing at the correct partition
(independently of the hardware revision).

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 rules/60-persistent-storage.rules | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rules/60-persistent-storage.rules b/rules/60-persistent-storage.rules
index 63f472be8..64053505e 100644
--- a/rules/60-persistent-storage.rules
+++ b/rules/60-persistent-storage.rules
@@ -56,6 +56,7 @@ KERNEL=="sd*[0-9]", ATTRS{ieee1394_id}=="?*", SYMLINK+="disk/by-id/ieee1394-$att
 KERNEL=="mmcblk[0-9]", SUBSYSTEMS=="mmc", ATTRS{name}=="?*", ATTRS{serial}=="?*", \
   ENV{ID_NAME}="$attr{name}", ENV{ID_SERIAL}="$attr{serial}", SYMLINK+="disk/by-id/mmc-$env{ID_NAME}_$env{ID_SERIAL}"
 KERNEL=="mmcblk[0-9]p[0-9]*", ENV{ID_NAME}=="?*", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/mmc-$env{ID_NAME}_$env{ID_SERIAL}-part%n"
+KERNEL=="mmcblk[0-9]p[0-9]", ENV{PARTNAME}=="?*", SYMLINK+="disk/by-partlabel/$env{PARTNAME}"
 
 # Memstick
 KERNEL=="msblk[0-9]|mspblk[0-9]", SUBSYSTEMS=="memstick", ATTRS{name}=="?*", ATTRS{serial}=="?*", \
